app/src/main/java/com/github/se/project/ui/lesson/CompletedLesson.kt

```diff
@@ -1,3 +1,219 @@
+    Text(
+        text = "Error. Should not happen",
+        style = MaterialTheme.typography.bodyLarge,
+        modifier = Modifier.fillMaxSize().padding(16.dp).testTag("errorText"), // Added test tag
+        color = MaterialTheme.colorScheme.error)
+  } else {
+    val otherProfile =
+        if (currentUserProfile!!.role == Role.STUDENT) {
+          listProfilesViewModel.getProfileById(lesson!!.tutorUid.firstOrNull()!!)
+        } else {
+          listProfilesViewModel.getProfileById(lesson!!.studentUid)
+        }
+
+    Scaffold(
+        modifier = Modifier.testTag("completedLessonScreen"), // Added test tag
+        topBar = {
+          TopAppBar(
+              title = { Text("Lesson Details", style = MaterialTheme.typography.titleLarge) },
+              navigationIcon = {
+                IconButton(
+                    onClick = { navigationActions.goBack() },
+                    modifier = Modifier.testTag("backButton") // Added test tag
+                    ) {
+                      Icon(Icons.Default.ArrowBack, contentDescription = "Go Back")
+                    }
+              },
+          )
+        }) { paddingValues ->
+          Column(
+              modifier =
+                  Modifier.padding(paddingValues)
+                      .fillMaxSize()
+                      .padding(horizontal = 16.dp, vertical = 8.dp)
+                      .testTag("completedLessonContent"), // Added test tag
+              verticalArrangement = Arrangement.spacedBy(16.dp)) {
+                if (otherProfile != null) {
+                  // give a bit more space
+                  Spacer(Modifier.padding(2.dp))
+
+                  // Role Information
+                  Text(
+                      text =
+                          "${if (otherProfile.role == Role.TUTOR) "Tutor" else "Student"} and lesson information",
+                      style = MaterialTheme.typography.titleMedium,
+                      color = MaterialTheme.colorScheme.primary,
+                      modifier = Modifier.testTag("roleInformation") // Added test tag
+                      )
+
+                  // Lesson Details
+                  DisplayLessonDetails(
+                      lesson = lesson!!,
+                      profile = otherProfile,
+                      modifier = Modifier.testTag("lessonDetails") // Pass modifier if possible
+                      )
+
+                  // give a bit more space
+                  Spacer(Modifier.padding(8.dp))
+
+                  // Review Header
+                  Text(
+                      text =
+                          if (currentUserProfile!!.role == Role.TUTOR) "Student's Review"
+                          else "Your Review",
+                      style = MaterialTheme.typography.titleMedium,
+                      color = MaterialTheme.colorScheme.primary,
+                      modifier = Modifier.testTag("reviewHeader") // Added test tag
```

Comment:
The readability of this section is excellent, thanks to the few comments and the good structure of your code.